services/navigation/builtin/builtin.go

```diff
@@ -88,49 +101,64 @@ type Config struct {
 func (conf *Config) Validate(path string) ([]string, error) {
 	var deps []string
 
+	// Add base dependencies
 	if conf.BaseName == "" {
 		return nil, utils.NewConfigValidationFieldRequiredError(path, "base")
 	}
 	deps = append(deps, conf.BaseName)
 
-	if conf.MovementSensorName == "" {
-		return nil, utils.NewConfigValidationFieldRequiredError(path, "movement_sensor")
+	// Add movement sensor dependencies
+	if conf.MovementSensorName != "" {
+		deps = append(deps, conf.MovementSensorName)
 	}
-	deps = append(deps, conf.MovementSensorName)
 
-	if conf.MotionServiceName == "" {
-		conf.MotionServiceName = resource.DefaultServiceName
+	// Add motion service dependencies
+	if conf.MotionServiceName != "" {
+		deps = append(deps, resource.NewName(motion.API, conf.MotionServiceName).String())
+	} else {
+		deps = append(deps, resource.NewName(motion.API, resource.DefaultServiceName).String())
 	}
-	deps = append(deps, resource.NewName(motion.API, conf.MotionServiceName).String())
 
+	// Add vision service dependencies
 	for _, v := range conf.VisionServices {
 		deps = append(deps, resource.NewName(vision.API, v).String())
 	}
 
-	// get default speeds from config if set, else defaults from nav services const
-	if conf.Store.Validate(path) != nil {
-		conf.Store.Type = defaultStoreType
+	// Ensure map_type is valid and a movement sensor is available if MapType is GPS (or default)
+	mapType, err := navigation.StringToMapType(conf.MapType)
+	if err != nil {
+		return nil, err
+	}
+	if mapType == navigation.GPSMap && conf.MovementSensorName == "" {
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "movement_sensor")
+	}
+
+	// Ensure store is valid
+	if err := conf.Store.Validate(path); err != nil {
+		return nil, err
 	}
-	if conf.MetersPerSec == 0 {
-		conf.MetersPerSec = defaultLinearVelocityMPerSec
+
+	// Ensure inputs are non-negative
+	if conf.DegPerSec < 0 {
+		return nil, errors.New("degs_per_sec must be non-negative if set")
 	}
-	if conf.DegPerSec == 0 {
-		conf.DegPerSec = defaultAngularVelocityDegsPerSec
+	if conf.MetersPerSec < 0 {
```

Comment:
[note] Previous no testing existed for checking if negative values were given for these parameters, should I add these in this PR?

Reply:
Always nice to have code improvements as a driveby :)